### PR TITLE
fix: squads ssr when referring user is present

### PR DESCRIPTION
**Pull request by sshanzel**
## Events

Did you introduce any new tracking events?

## Experiment

Did you introduce any new experiments?

## Manual Testing

> [!CAUTION]
> Please make sure existing components are not breaking/affected by this PR

### Preview domain
https://fix-squads-ssr.preview.app.daily.dev

### CODE REVIEW

File: packages/webapp/pages/squads/[handle]/index.tsx [hunk 1]
```diff
@@ -286,6 +286,7 @@ export async function getServerSideProps({
               id: userId,
             },
           )
+          .then((data) => data?.user)
```
Comment (capJavert) [inline packages/webapp/pages/squads/[handle]/index.tsx:289]: below you have `referringUser: referringUser?.user || null` so it needs to be adjusted

**Review by capJavert — CHANGES_REQUESTED** [review]

**Review by idoshamun — APPROVED** [review]

File: packages/webapp/pages/squads/[handle]/index.tsx (hunk 1 above)
Comment (rebelchris) [inline packages/webapp/pages/squads/[handle]/index.tsx:289]: Wait we just missed this 😂

**Review by rebelchris — APPROVED** [review]

File: packages/webapp/pages/squads/[handle]/index.tsx (hunk 1 above)
Comment (sshanzel) [reply packages/webapp/pages/squads/[handle]/index.tsx:289]: Yes 😂

Comment (sshanzel) [reply packages/webapp/pages/squads/[handle]/index.tsx:289]: Right. Let me update it quickly.

Comment (sshanzel) [reply packages/webapp/pages/squads/[handle]/index.tsx:290]: To ensure we won't miss anything, I applied some strict types.

**Review by capJavert — APPROVED** [review]

**Pull request merged**